### DISCUSSION TITLE
use table body-cell slot for grid mode

### DISCUSSION THREAD

**Discussion by ryhn** [quasarframework/quasar#15599] [Ideas / Proposals] [answered]
while using grid-mode, template body-cell-name not inherit to grid mode. All code needs to be added again. it would be great if the body-cell slot could use

**Comment by saike** [comment]
you should use 'v-slot:item' when you using the 'grid' param

**Comment by metalsadman** [comment, marked as answer]
That's better left in devland the slots is already available, however you can use `component` and a qtable wrapper with the implementation to use the same body-cell slots for when changing to grid or vice versa. Here https://codepen.io/metalsadman/pen/ZgKexK
See my comments in the code.

```html
   <template #body-cell-status="props">
      <component :is="grid ? 'div' : 'q-td'" :props="props">
        <q-chip
          :color="props.row.status === 'Active' ? 'green': 'red'"
          text-color="white"
          dense
          class="text-weight-bolder"
          square
        >{{props.row.status}}</q-chip>
      </component>
  </template>
```

**Reply by ryhn** [reply]
@metalsadman your code is good solution. Thanks.
